service/history/replicationTaskProcessor.go

```diff
@@ -278,6 +364,32 @@ func isTransientRetryableError(err error) bool {
 	}
 }
 
+func (p *ReplicationTaskProcessor) shouldRetryDLQ(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	select {
+	case <-p.done:
+		p.logger.Info("ReplicationTaskProcessor shutting down.")
+		return false
+	default:
+		return true
+	}
+}
+
+func toPersistenceReplicationInfo(info map[string]*shared.ReplicationInfo) map[string]*persistence.ReplicationInfo {
```

Comment:
should the kafka deprecation support 2DC case?

Reply:
also format plz

Reply:
It should support 2DC case. Anything here you think that does not work for 2DC?